$MirOS: ports/devel/glib/patches/patch-glib_gmacros_h,v 1.3 2008/03/06 22:08:24 tg Exp $
$OpenBSD: patch-glib_gmacros_h,v 1.3 2004/06/22 03:15:02 marcm Exp $
--- glib/gmacros.h.orig	Fri Sep 12 19:55:28 2008
+++ glib/gmacros.h	Thu Nov 25 12:24:44 2010
@@ -34,6 +34,7 @@
 /* We include stddef.h to get the system's definition of NULL
  */
 #include <stddef.h>
+#include <sys/param.h>
 
 /* Here we provide G_GNUC_EXTENSION as an alias for __extension__,
  * where this is valid. This allows for warningless compilation of
@@ -171,11 +172,13 @@
 #define	TRUE	(!FALSE)
 #endif
 
-#undef	MAX
+#ifndef MAX
 #define MAX(a, b)  (((a) > (b)) ? (a) : (b))
+#endif
 
-#undef	MIN
+#ifndef MIN
 #define MIN(a, b)  (((a) < (b)) ? (a) : (b))
+#endif
 
 #undef	ABS
 #define ABS(a)	   (((a) < 0) ? -(a) : (a))
